src/sql/workbench/parts/objectExplorer/browser/objectExplorerViewTreeShim.ts

```diff
@@ -97,6 +97,10 @@ export class OEShimService extends Disposable implements IOEShimService {
 				onConnectStart: undefined,
 				onDisconnect: undefined
 			});
+			// connection cancelled from firewall dialog
+			if (!result) {
+				reject(new UserCancelledConnectionError(localize('loginCanceled', "User canceled")));
+			}
```

Comment:
So it looks like this is going to call onConnectReject if the firewall rule fails to connect. If so that seems like the more correct place to do this (like in onConnectCanceled)

[https://github.com/microsoft/azuredatastudio/blob/ea0f9e6ce95b45def2b694efcd8ffe909bc43d20/src/sql/platform/connection/browser/connectionManagementService.ts#L450](https://github.com/microsoft/azuredatastudio/blob/ea0f9e6ce95b45def2b694efcd8ffe909bc43d20/src/sql/platform/connection/browser/connectionManagementService.ts#L450)

Reply:
The api has no onConnectReject by default, and there's no way to supply one. I don't want to change the API, so I'll let this be.